Calling two() method in flask util test.

dhermes opened the pull request:
In the original code, the line never executes since it is in a method that is decorated with `oauth2.required` and the method's scopes don't match the credentials used (forcing a redirect). Thus, we call it again with credentials that actually match the scopes.

dhermes commented:
@nathanielmanistaatgoogle After this, the only line coverage [missing in tests][1] are `self.fail()` lines.

Towards #212 inch-by-inch!

[1]: https://github.com/google/oauth2client/pull/297#issuecomment-142150899

dhermes commented:
@jonparrott FYI I figured this was the easiest fix.

In general I do my best to avoid `NO COVER` pragmas. If you have a better idea for how to get this line actually executed. LMK.

jonparrott commented:
You could make another request to `/two` with all the valid scopes, but it seems like overkill.

dhermes commented:
I'll let @nathanielmanistaatgoogle decide.

Now that you point that out, it seems it'd be as easy as:

```python
        credentials2 = self._generate_credentials(scopes=['two'])

        with self.app.test_client() as c:
            with c.session_transaction() as session:
                session['google_oauth2_credentials'] = credentials2.to_json()

            rv = c.get('/two')
            self.assertEqual(rv.status_code, httplib.OK)
```

nathanielmanistaatgoogle commented:
Let's go with the latter approach that is a handful of lines of more code and avoids the coverage suppression pragma.

dhermes committed:
Calling two() method in flask util test.

In the original code, the line never executes since it is in
a method that is decorated with oauth2.required and the
method's scopes don't match the credentials used (forcing a
redirect). Thus, we call it again with credentials that actually
match the scopes.

dhermes commented:
@nathanielmanistaatgoogle Updated the commit (and the messaging on the PR).
